app/presenters/speech_presenter.rb

```diff
@@ -0,0 +1,41 @@
+class SpeechPresenter < ContentItemPresenter
+  def delivery_type
+    if content_item.document_type == "authored_article"
+      I18n.t("formats.speech.written_on")
+    else
+      I18n.t("formats.speech.delivered_on")
+    end
+  end
+
+  def delivered_on_metadata
+    "#{delivered_on}#{speech_type_explanation}"
+  end
+
+  def important_metadata
+    { I18n.t("formats.speech.location") => location, delivery_type => delivered_on_metadata }
+  end
+
+  def view_context
+    @view_context = ApplicationController.new.view_context
+  end
+
+  def display_date(timestamp, format = "%-d %B %Y")
```

Comment:
You can remove this method as the helper method is already available in [date_helper.rb](https://github.com/alphagov/frontend/blob/main/app/helpers/date_helper.rb)